bindings/python/src/asyncio.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use databend_client;
+use databend_driver::rest_api::RestAPIConnection;
+use pyo3::prelude::*;
+use pyo3_asyncio::tokio::future_into_py;
+use databend_driver::Connection;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{Arc, RwLock},
+};
+use std::fs::Metadata;
+use pyo3::pyobject_native_type;
+
+use crate::{build_rest_api_client};
+use crate::format_pyerr;
+
+/// `AsyncDatabendDriver` is the entry for all public async API
+#[pyclass(module = "databend_python")]
+pub struct AsyncDatabendDriver(RestAPIConnection);
```

Comment:
use trait for both instead of `RestAPIConnection`?
```suggestion
pub struct AsyncDatabendDriver(Box<dyn Connection>);
```

Reply:
I have not found a way to compile it when using trait `Connection`. 😢

Reply:
I will try it again today